examples/cis2-multi-royalties/src/lib.rs

```diff
@@ -0,0 +1,1405 @@
+) -> ContractResult<SupportsQueryResponse> {
+    // Parse the parameter.
+    let params: SupportsQueryParams = ctx.parameter_cursor().get()?;
+
+    // Build the response.
+    let mut response = Vec::with_capacity(params.queries.len());
+    for std_id in params.queries {
+        if SUPPORTS_STANDARDS.contains(&std_id.as_standard_identifier()) {
+            response.push(SupportResult::Support);
+        } else {
+            response.push(host.state().have_implementors(&std_id));
+        }
+    }
+    let result = SupportsQueryResponse::from(response);
+    Ok(result)
+}
+
+/// Set the addresses for an implementation given a standard identifier and a
+/// list of contract addresses.
+///
+/// It rejects if:
+/// - Sender is not the owner of the contract instance.
+/// - It fails to parse the parameter.
+#[receive(
+    contract = "cis2_multi_royalties",
+    name = "setImplementors",
+    parameter = "SetImplementorsParams",
+    error = "ContractError",
+    mutable
+)]
+fn contract_set_implementor<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &mut impl HasHost<State<S>, StateApiType = S>,
+) -> ContractResult<()> {
+    // Authorize the sender.
+    ensure!(ctx.sender().matches_account(&ctx.owner()), ContractError::Unauthorized);
+    // Parse the parameter.
+    let params: SetImplementorsParams = ctx.parameter_cursor().get()?;
+    // Update the implementors in the state
+    host.state_mut().set_implementors(params.id, params.implementors);
+    Ok(())
+}
+
+/// Returns the address of the payee and the amount of royalties to pay them
+/// 
+/// It rejects if:
+/// - The contract does not pay royalties
+/// - The return result would be 0
+/// - The token is not recognised
+#[receive(
+    contract = "cis2_multi_royalties",
+    name = "check_royalty",
+    parameter = "CheckRoyaltyParams",
+    error = "ContractError",
+    mutable,    
+)]
+fn contract_check_royalty<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &mut impl HasHost<State<S>, StateApiType = S>,
+) -> ContractResult<CheckRoyaltyResult> {
+    ensure!(host.state().pay_royalty == true, concordium_cis2::Cis2Error::Custom(CustomContractError::ContractDoesNotPayRoyalties));
+
+    let params: CheckRoyaltyParams = ctx.parameter_cursor().get()?;
+    ensure!(params.sale_price > 0, concordium_cis2::Cis2Error::Custom(CustomContractError::NoRoyaltyPayable));
+    ensure!(host.state().contains_token(&params.id), ContractError::InvalidTokenId);
+    let token_details = host.state().token_details.get(&params.id).unwrap(); // safe due to previous checks
+    ensure!(token_details.royalty > 0, concordium_cis2::Cis2Error::Custom(CustomContractError::NoRoyaltyPayable));
+    
+    Ok(host.state_mut().check_royalty_payments(params))
+}
+
+/// View function to see whether we support paying royalties
+#[receive(contract = "cis2_multi_royalties", name = "supports_interface", return_value = "bool")]
+fn supports_interface<S: HasStateApi>(
```

Comment:
```suggestion
fn contract_pays_royalties<S: HasStateApi>(
```
Consider using a more descriptive name.

Reply:
@NossicaSystems Please address or comment the current naming makes sense as well.

Reply:
Updated